### PR TITLE
Init bootstrap gradle plugin

### PR DESCRIPTION
**Pull request by glefloch**
This branch introduce a new Gradle boostrap plugin that will help extension developer if they want to use gradle.
This is a first step that only generate the `quarkus-extension.properties` file and does not handle capabilities for now.

One issue I see so far, is that the group id of the project corresponds to the plugin name, in that case `io.quarkus.bootstrap`. 
Can that be an issue when publishing artifacts?

### CODE REVIEW

File: independent-projects/bootstrap/gradle-plugin/gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -0,0 +1,9 @@
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+<<<<<<< HEAD
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.9-all.zip
+=======
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.8-bin.zip
+>>>>>>> d232aad03c (Add wrapper)
```
Comment (aloubyansky) [inline independent-projects/bootstrap/gradle-plugin/gradle/wrapper/gradle-wrapper.properties:7]: This looks like a leftover from a merge?

**Comment by aloubyansky** [comment]
I am not sure bootstrap is the right place for it though. Which is also true for the corresponding Maven goal. It's related to the extension building. Perhaps it should be `io.quarkus.extension` plugin or something.

File: independent-projects/bootstrap/gradle-plugin/gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (glefloch) [reply independent-projects/bootstrap/gradle-plugin/gradle/wrapper/gradle-wrapper.properties:7]: Oh right, this is a mistake.

**Comment by glefloch** [comment]
@aloubyansky, I fixed the conflict and renamed the plugin. Should I move it somewhere else or is `independent-projects/bootstrap` ok for now?

**Comment by aloubyansky** [comment]
Perhaps we could make https://github.com/quarkusio/quarkus/tree/main/devtools/gradle a common parent for the `io.quarkus` and `io.quarkus.extension` plugins?

**Comment by aloubyansky** [comment]
It could probably be a multi-module gradle project?

**Comment by aloubyansky** [comment]
Not sure though whether it helps from the release/publishing perspective.

**Comment by aloubyansky** [comment]
We'll need to start some "Quarkus extension development with Gradle" docs too :)

**Comment by glefloch** [comment]
We could move it under ˋdevtools/gradle` but not sure if it helps.. 
regarding docs, I wanted that the plug-in was able to write both descriptor file before writing it. 
WDYT?

**Comment by aloubyansky** [comment]
I thought that way both plugins could share a common parent config and would be built and released at the same time. Otherwise, if they are two independent projects they would have to be released kind of separately. Or is it not exactly true? @gsmet, wdyt?

As to docs, sure, they can take time.

**Comment by glefloch** [comment]
Yes I think that would make sense to release both plugin at the same time. And the gradle part would be only in one place.

**Comment by glefloch** [comment]
@aloubyansky I created a multi-module gradle project that contains both plugins.

File: devtools/gradle/application-plugin/build.gradle [hunk 2]
```diff
@@ -0,0 +1,27 @@
+dependencies {
+    api gradleApi()
+
+    implementation "io.quarkus:quarkus-bootstrap-core:${version}"
+    implementation "io.quarkus:quarkus-devtools-common:${version}"
+    implementation "io.quarkus:quarkus-core-deployment:${version}"
+
+    testImplementation "io.quarkus:quarkus-project-core-extension-codestarts:${version}"
+    testImplementation 'org.assertj:assertj-core:3.20.2'
+    testImplementation 'org.junit.jupiter:junit-jupiter:5.7.2'
```
Comment (aloubyansky) [inline devtools/gradle/application-plugin/build.gradle:10]: The above versions seem to be common for both plugins, could we define them in the parent project?

**Comment by aloubyansky** [comment]
@gsmet we should check how this will affect the release process/scripts.

File: devtools/gradle/application-plugin/build.gradle (hunk 2 above)
Comment (glefloch) [reply devtools/gradle/application-plugin/build.gradle:10]: oh right, fixed, with other common dependencies.

**Review by aloubyansky — APPROVED** [review]

**Review by gsmet — DISMISSED** [review]
For me, it's too big of a change for 2.2 which is a harderning release. I marked it as `waiting-for-2.3`.

**Comment by gsmet** [comment]
@glefloch @famod @aloubyansky I suggest we wait for the Gradle 7.2 upgrade to get in before rebasing and merging this one.

**Comment by glefloch** [comment]
@gsmet sure, I rebased my branch to check if everything works with gradle 7.

**Comment by glefloch** [comment]
@gsmet can we merge this one?

**Comment by aloubyansky** [comment]
@glefloch is it good to merge in your opinion?

**Comment by glefloch** [comment]
I think so, most annoying gradle issues have been a fix. 
However, this introduce a new artifact with a new group ˋio.quarkus.extension` I don’t know if this have any impact..

**Comment by aloubyansky** [comment]
@geoand I would merge it ASAP but I'd like to make sure we are prepared to release it. It is second Gradle plugin.

**Comment by aloubyansky** [comment]
@glefloch please at least a little doc describing how to use it and what it does. It could be in a separate PR.

**Comment by geoand** [comment]
Gotcha!

**Comment by geoand** [comment]
@gastaldi IIRC you handled the details of releasing the original plugin the first time around?

**Comment by glefloch** [comment]
@aloubyansky yes, I'm taking care of the doc today.

**Comment by geoand** [comment]
So we'll probably summon you next week :)

**Comment by gastaldi** [comment]
Publishing is easy, you can use the same token for the original plugin: https://github.com/quarkusio/quarkus-release/blob/main/release-gradle-plugin.sh#L24

**Comment by geoand** [comment]
We'll give it a shot :)

**Comment by glefloch** [comment]
I think this should still work and publish both plugin

**Comment by aloubyansky** [comment]
@glefloch little follow up request. Could you please add `gradle-` prefixes to the plugin Maven modules? So they appear under the `gradle-` prefix in the IDE.

**Pull request merged**